pkg/skaffold/runner/dev.go

```diff
@@ -50,11 +50,15 @@ var (
 )
 
 func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer, logger *kubernetes.LogAggregator, forwarderManager portforward.Forwarder) error {
+	// never queue intents from user, even if they're not used
+	defer r.intents.Reset()
+
 	if r.changeSet.needsReload {
 		return ErrorConfigurationChanged
 	}
 
 	buildIntent, syncIntent, deployIntent := r.intents.GetIntents()
+	logrus.Tracef("dev intents: build %t, sync %t, deploy %t\n", buildIntent, syncIntent, deployIntent)
```

Comment:
should we notify the user either in the response to the API request (but that'd require making it synchronous) or a string output that we're skipping this request since nothing has changed?

Reply:
this would require a bit more plumbing, since in the runner we're just asking whether the intent's gate is open, not whether the user requested to open it. i think this would be a good improvement, but it should probably happen outside the scope of this PR.